src/passes/I64ToI32Lowering.cpp

```diff
@@ -32,6 +32,8 @@
 
 namespace wasm {
 
+bool i64ToI32InjectFetchHighBits = false;
```

Comment:
It seems like this doesn't need to be in the i64 lowering pass? It could be directly in wasm2asm (where the global is set), avoiding the global.